idris-repl.el

```diff
@@ -186,9 +186,16 @@ If ALWAYS-INSERT is non-nil, always insert a prompt at the end of the buffer."
       (idris-repl-insert-prompt)
       (insert current-input))))
 
+(autoload 'idris-load-file "idris-commands.el")
+;;;###autoload
 (defun idris-switch-to-repl ()
   "Select the output buffer and scroll to bottom."
   (interactive)
+  (if (and buffer-file-name
+           ;; in Emacs 29.1 > we can use string-equal-ignore-case
+           (string= "idr" (downcase (file-name-extension buffer-file-name))))
```

Comment:
Is it not the case that idris files can have other extensions if we are in literate mode?

Reply:
Good point! I have updated the implementation to check also for `lidr` files although I don't have any good lidr examples to test it with.